### DISCUSSION TITLE
Does the tolerance here mean relative tolerance or absolute  tolerance?

### DISCUSSION THREAD

**Discussion by ORANGELCX** [OpenModelica/OpenModelica#9745] [Q&A] [answered]
### Description
When I use openmodelica to calculate the ode equation, I am very confused about this tolerance. I don't know if it refers to relative error or absolute tolerance.

### Screenshots
![DC2DF606-7D80-4F46-B305-C241B479AA62](https://user-images.githubusercontent.com/55785611/203492384-4d2d96d0-1639-41c4-a687-0de728d2637f.png)

**Comment by AnHeuermann** [comment, marked as answer]
It's relative tolerance and is used to compute absolute tolerance.

The default explicit Euler method doesn't use this tolerance, since it doesn't has step-size control. If you want to use Euler with step size control and therefore tolerance, check the [GBODE solver](https://www.openmodelica.org/doc/OpenModelicaUsersGuide/latest/solving.html#gbode). 

For some solvers (e.g. DASSL, the default solver) the user provided relative tolerance is also used to compute the absolute tolerance for each state:
```math
tol_{abs}[i] = tol_{rel} \cdot nominal[i]
```

**Reply by ORANGELCX** [reply]
nominal[i] refers to？

**Reply by AnHeuermann** [reply]
The nominal value of the ith state.

**Reply by casella** [reply]
It's the nominal attribute, like in 
```modelica
  Real x(min = 1e8, max = 2e8, nominal = 1e8, start = 1.5e8);
```
Nominal attributes provide an order of magnitude that can be used for scaling purposes. This is especially useful for variables that may have values of zero at some point. For example, if you have a power controller for a 100 MW power plant, when the power is equal to the set point, the error variable will be zero. However, its order of magnitude is 1e8 (100,000,000 W in SI units), so a "small value" won't be, say 1e-12, but rather 1e-12*nominal = 1e-4.

**Reply by casella** [reply]
See also http://dx.doi.org/10.1145/3158191.3158192

**Reply by ORANGELCX** [reply]
@AnHeuermann @casella I have understood, and thank you both for your replies
